sdks/python/apache_beam/typehints/typehints.py

```diff
@@ -462,7 +462,7 @@ def match_type_variables(self, concrete_type):
     return {self: concrete_type}
 
   def bind_type_variables(self, bindings):
-    return bindings.get(self, self)
+    return bindings.get(self, bindings.get('*', self))
```

Comment:
Maybe add a comment explaining what the special key `'*'` means, i.e., that it will apply to all unbound type variables.

Reply:
Good point. Done.